File: apigateway/src/main/java/ru/otus/apigateway/model/backend/BasketItem.java
package ru.otus.apigateway.model.backend;

import lombok.Builder;
import lombok.Data;
import ru.otus.apigateway.model.view.SubscriptionViewModel;

@Data
@Builder
public class BasketItem {

    private Long id;
    private Integer quantity;
    private Long customerId;
    private Long subscriptionId;

    private SubscriptionViewModel subscription;
}

File: frontend/src/app/main-module/models/basket-item.ts
import {Subscription} from "./subscription";

export class BasketItem {

    id: string;
    customerId: string;
    subscription: Subscription = new Subscription();
    quantity: number;

    constructor(customerId: string, subscription: Subscription, quantity: number) {
        this.customerId = customerId;
        this.subscription = new Subscription();
        this.subscription.id = subscription.id;
        this.quantity = quantity;
    }
}


